app/jobs/send_purge_warnings_job.rb

```diff
@@ -9,13 +9,13 @@ def perform(*_args)
     not_sent = []
 
     # Get admin users
-    recipients = User.with_any_role(:admin)
+    recipients = User.where(role: Roles::ADMIN) | User.where(role: Roles::SUPER_USER)
     eligible = recipients.count
 
     # Loop through and send each admin information about their purge eligible monitorees
     recipients.each do |user|
       # Skip for USA admins
-      next if user.jurisdiction&.name == 'USA'
+      next unless user.can_receive_purge_warning?
```

Comment:
I actually think this was fine as is. With this, it has to re-check the roles which isn't necessary so I'd say you can go back to what was there before.